Handle archives and preferred content better

matejak opened the pull request:
The addon lost the ability to pick files from supplied archives based on kickstart specifications.
This PR allows this, which allows the addon to

- handle archives with e.g. multiple datastream files, and to
- throw error when an expected file in the archive is missing.

This PR fixes RHEL8 and RHEL9 BZs

- https://bugzilla.redhat.com/show_bug.cgi?id=2129008
- https://bugzilla.redhat.com/show_bug.cgi?id=2129846

matejak committed:
Add capability to preselect content from archives

Users can specify content path and tailoring path in kickstarts,
and the addon should be able to assure that those files are available,
and that they have precedence over other files.

matejak committed:
Remove unused code

The function is not referenced anywhere in the project

matejak committed:
Handle changes in content identification

The code is able to handle changes in the way how oscap identifies
content much more gracefully.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/59735a2d-2b08-477e-8acb-ba5d9f454b33) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/59735a2d-2b08-477e-8acb-ba5d9f454b33) completed: **2 new issues, 5 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/467631eb-8f9d-4b21-8646-888b3ef67d40) was created.

jan-cerny commented:
@matejak I find it difficult to review this PR.

1. I need to reproduce the BZs to verify that the changes in this PR fix the BZs. However, I miss a kickstart that is used to reproduce this BZs. When I add the %addon org_fedora_oscap snippet from rhbz#2129008 to my kickstart, change the IPs, URLs and paths in that snippet for my local values and then start a RHEL 8.7 installation, I get "error when fetching content" in the OAA spoke in the bottom, which is probably fine, but I don't see the messages that are listed in the BZ in the /tmp/anaconda.log. I assume that I can similarly struggle to reproduce with RHEL 9.1. Therefore, please attach the kickstarts that you used to reproduce and test this BZ on both RHEL 8.7 and 9.1 to this PR. Also, please describe the steps that you performed when you tested your patch.

2. This PR is against master branch which is on Fedora, but I don't know how to test Fedora as the linked BZs are on RHEL 8.7 and 9.1. When I try to download the patch file from this PR URL and apply it to current upstream rhel8-branch branch it doesn't apply, probably there are conflicts. I can try to resolve these locally but I don't know if this would be fine. How can I test that this PR fixes the linked RHEL BZs?

matejak committed:
Dont use tailoring if it is not expected

Take tailorings into account only if it is specified in the kickstart.
Compulsive usage of tailoring may be unwanted.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/97521932-8423-4448-8967-e1799d80086c) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/97521932-8423-4448-8967-e1799d80086c) completed: **6 updated code elements**

jan-cerny commented:
To clarify, this will be done after https://github.com/OpenSCAP/oscap-anaconda-addon/pull/220 will be discussed because it's a RHEL 8 variant of this PR.

jan-cerny commented:
@matejak The PR #220 has been merged so this PR is now unblocked and you can start working on it.

Also, can you answer the question above about testing on Fedora? What are the steps that you have taken when testing this on Fedora?

jan-cerny commented:
@matejak I'm strongly disappointed by no response.

jan-cerny commented:
@matejak PING

jan-cerny commented:
@matejak ping
